oaknational/Oak-Web-Application
# Feat/lesq 11/downloads layout [LESQ-11]

## Description

- Deleted old download card
- Renamed DLCard to `ResourceCard` so it can be used for share
- made a new component `ResourcePageLayout` to handle the layout for download & share pages
- added some logic to organise the resouce cards according to ACS
- moved some components from the `LessonDownloads.page` into `ResourcePageLayout`


## How to test

I've added a few test cases in storybook on the `DownloadCardGroup` component to cover the different scenarios of resources present (with & without presentation, with one/two worksheets) : https://deploy-preview-2064--owa-storybook.netlify.thenational.academy/?path=/story/download-components-download-card-group--all-resources 

1. Go to https://deploy-preview-2064--oak-web-application.netlify.thenational.academy/teachers/programmes/combined-science-secondary-ks4-foundation-ocr/units/measuring-waves/lessons/transverse-waves/downloads?preselected=all
3. You should see the new layout
4. Go to https://deploy-preview-2064--oak-web-application.netlify.thenational.academy/teachers/programmes/religious-education-secondary-ks4-l/units/islamic-practices-690d/lessons/shahadah-6gu68c/downloads?preselected=all
5. See fewer resources

## Screenshots

How it used to look (delete if n/a):
<img width="512" alt="Screenshot 2023-11-01 at 09 20 45" src="https://github.com/oaknational/Oak-Web-Application/assets/45038878/a422a5b7-441f-4307-953d-5f329c2cd93d">


How it should now look:


<img width="512" alt="Screenshot 2023-11-01 at 09 30 02" src="https://github.com/oaknational/Oak-Web-Application/assets/45038878/e7f3db16-c5bb-4471-b4f6-12b83f176db5">

## ACs

- [ ]  Column x 2 format (resources on left and details on right)
- [ ]  Swap to new resource selector component
    - Slide deck appears on its own row
    - Unless there is only 1 worksheet resource, in which case that buddies up with the slide deck on the top row
    - Everything else pairs with its no answer/answer variant
    - Everything just moves up if there is no slide deck
- [ ]  Remove old download card component
- [ ]  Reposition t’s & c’s, copyright details and download button
- [ ]  Heading changes - check figma notes
